packages/grid/x-grid-modules/src/components/error-boundary.tsx

```diff
@@ -0,0 +1,44 @@
+import * as React from 'react';
+import { ErrorInfo } from 'react';
+import { Logger } from '../hooks/utils/useLogger';
+import { ApiRef } from '../models/api';
+
+export interface ErrorBoundaryProps {
+  logger: Logger;
+  render: ({ error }: any) => React.ReactNode;
+  api: ApiRef;
+  hasError: boolean;
+  componentProps?: any[];
+}
+
+export class ErrorBoundary extends React.Component<ErrorBoundaryProps, {}> {
+  constructor(props: any) {
+    super(props);
+  }
+
+  componentDidCatch(error: Error, errorInfo: ErrorInfo) {
+    if (this.props.api.current) {
+      this.logError(error);
+
+      // Allows to trigger the Error event and all listener can run on Error
+      this.props.api.current.showError({ error, errorInfo });
```

Comment:
Would this work? I doubt that a parent Error boundary would catch the error but

```suggestion
      // Allows to trigger the Error event and all listener can run on Error
      this.props.api.current.showError({ error, errorInfo });

      // Rethrow the error so tracking tools can collect it automatically
      // without having to manually listen to the onError callback.
      throw error;
```

Reply:
Yes it's working as intended. Error is thrown and error component is displayed. No need of that ;)

Reply:
Let's say I was a developer looking for an advanced grid. If the grid crash, what default would I want? I can think of two cases:

- I would freeze the UI, add a reset button, and keep allowing end-users to read the content.
- Alternatively, I would turn the whole app with a white screen, a message to contact support, and a reload button.

Reply:
> How does it work? In this case: https://codesandbox.io/s/affectionate-stonebraker-m7t0f?file=/src/App.js, the error is trapped at the "grid" level, it can never escape. The only way seems to use an `onError` callback (that needs to be memorized) but it breaks the pattern of error boundaries.

Not sure what you mean here. The codesandbox use the react-boundary package, I use the native react error boundary

> If the grid crash, what default would I want? I can think of two cases:

Well the behaviour of the app is subject to the consumer, but what you don't want, is to show a corrupt component or corrupt data.

Reply:
> The codesandbox use the react-boundary package

Fair enough, using the React primitives directly: https://codesandbox.io/s/affectionate-stonebraker-m7t0f?file=/src/App.js, same problem :).

> what you don't want, is to show a corrupt component or corrupt data.

Agree, I wonder how much corrupted the displayed data can become.

Reply:
What was the conclusion? Should we tell people in the documentation that errors are swallowed in the error boundary, that they can't use the pattern with the grid, but use the `onError` listener instead? https://codesandbox.io/s/affectionate-stonebraker-m7t0f?file=/src/App.js